Code is broken when compile under linux (Fedora 27)

Hi,
this is a bug https://github.com/vbextreme/RigsOfRodsFedora/issues/1 because the code is broken and not compile under linux.
i'm here for more information.

I see the problem here:
```
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/configurator/SystemTools.cpp: In function ‘bool ExtractZipFiles(const wxString&, const wxString&)’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/configurator/SystemTools.cpp:148:49: error: ‘wxPosixPermissions’ has not been declared
             if (! wxFileName::Mkdir(subdirPath, wxPosixPermissions::wxS_DIR_DEFAULT, wxPATH_MKDIR_FULL))
                                                 ^~~~~~~~~~~~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/configurator/SystemTools.cpp:156:70: error: invalid conversion from ‘const char*’ to ‘wxChar {aka wchar_t}’ [-fpermissive]
         if ((lastSepPos != -1) && (entryName.substr(lastSepPos+1) == "empty") && (entry->GetSize() == 0))
    
```
Looks like WxWidgets version issue

@vbextreme with new script the build go to same error
```
[vluongo@vluongo-fedora-hpo RigsOfRodsFedora]$ sudo ./ror.build stable
Switched to a new branch 'fedora-stable-0'
HEAD ora si trova a e9edfb7c :bug: Fixed building without USE_SOCKETW
Si è passati al branch 'master'
Your branch is up-to-date with 'origin/master'.
[vluongo@vluongo-fedora-hpo RigsOfRodsFedora]$ cd rigs-of-rods/
[vluongo@vluongo-fedora-hpo rigs-of-rods]$ sudo su
[root@vluongo-fedora-hpo rigs-of-rods]# rm -f CMakeCache.txt
[root@vluongo-fedora-hpo rigs-of-rods]# find . -name CmakeFiles -exec rm -rf {} \;
[root@vluongo-fedora-hpo rigs-of-rods]# cmake -DwxWidgets_wxrc_EXECUTABLE=/usr/bin/wxrc-2.8 -DwxWidgets_CONFIG_EXECUTABLE=/usr/bin/wx-config-2.0 .
-- Found PythonInterp: /bin/python3 (found suitable version "3.6.3", minimum required is "3.0") 
-- The C compiler identification is GNU 7.2.1
-- The CXX compiler identification is GNU 7.2.1
-- Check for working C compiler: /bin/cc
-- Check for working C compiler: /bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /bin/c++
-- Check for working CXX compiler: /bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /bin/pkg-config (found version "1.3.12") 
-- Checking for module 'gtk+-2.0'
--   Found gtk+-2.0, version 2.24.31
-- Checking for module 'gdk-pixbuf-2.0'
--   Found gdk-pixbuf-2.0, version 2.36.11
-- Found Git: /bin/git (found version "2.14.3") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Looking for OGRE...
-- OGRE_PREFIX_WATCH changed.
-- Checking for module 'OGRE'
--   Found OGRE, version 1.9.0
-- Found Ogre Ghadamon (1.9.0)
-- Found OGRE: /usr/lib64/libOgreMain.so
-- Looking for OGRE_Paging...
-- Found OGRE_Paging: /usr/lib64/libOgrePaging.so
-- Looking for OGRE_Terrain...
-- Found OGRE_Terrain: /usr/lib64/libOgreTerrain.so
-- Looking for OGRE_Property...
-- Found OGRE_Property: /usr/lib64/libOgreProperty.so
-- Looking for OGRE_RTShaderSystem...
-- Found OGRE_RTShaderSystem: /usr/lib64/libOgreRTShaderSystem.so
-- Looking for OGRE_Volume...
-- Found OGRE_Volume: /usr/lib64/libOgreVolume.so
-- Looking for OGRE_Overlay...
-- Found OGRE_Overlay: /usr/lib64/libOgreOverlay.so
-- Looking for OIS...
-- OIS_PREFIX_PATH changed.
-- Checking for module 'OIS'
--   Found OIS, version 1.3.0
-- Found OIS: /usr/lib64/libOIS.so
-- Found MyGUI: /usr/local/include/MYGUI  
-- Found OpenAL: /usr/lib64/libopenal.so  
-- Found SocketW: /usr/local/include  
-- Found AngelScript: /usr/local/include  
-- Found CURL: /usr/lib64/libcurl.so (found version "7.55.1") 
-- Found Caelum: /usr/local/include/Caelum  
-- Could NOT find PagedGeometry (missing: PagedGeometry_INCLUDE_DIRS PagedGeometry_LIBRARIES) 
CMake Warning at source/main/CMakeLists.txt:511 (message):
  Building RoR without PagedGeometry.  Some functionality may be missing.


-- Found wxWidgets: -pthread;-Wl,-z,relro;-specs=/usr/lib/rpm/redhat/redhat-hardened-ld;;-lwx_baseu-2.8;-lwx_gtk2u_core-2.8;-lwx_gtk2u_html-2.8;-lwx_baseu_net-2.8;-lwx_gtk2u_adv-2.8 (found version "2.8.12") 
-- Looking for OGRE...
-- Found Ogre Ghadamon (1.9.0)
-- Found OGRE: /usr/lib64/libOgreMain.so
-- Looking for OGRE_Paging...
-- Found OGRE_Paging: /usr/lib64/libOgrePaging.so
-- Looking for OGRE_Terrain...
-- Found OGRE_Terrain: /usr/lib64/libOgreTerrain.so
-- Looking for OGRE_Property...
-- Found OGRE_Property: /usr/lib64/libOgreProperty.so
-- Looking for OGRE_RTShaderSystem...
-- Found OGRE_RTShaderSystem: /usr/lib64/libOgreRTShaderSystem.so
-- Looking for OGRE_Volume...
-- Found OGRE_Volume: /usr/lib64/libOgreVolume.so
-- Looking for OGRE_Overlay...
-- Found OGRE_Overlay: /usr/lib64/libOgreOverlay.so
-- Looking for OIS...
-- Found OIS: /usr/lib64/libOIS.so
-- The following OPTIONAL packages have been found:

 * Git
 * SocketW
 * CURL
 * Caelum
 * AngelScript
 * PkgConfig
 * OpenAL

-- The following REQUIRED packages have been found:

 * PythonInterp (required version >= 3.0)
 * wxWidgets
 * Threads
 * OGRE
 * OIS
 * MyGUI

-- The following OPTIONAL packages have not been found:

 * PagedGeometry

-- Configuring done
-- Generating done
-- Build files have been written to: /home/vluongo/RigsOfRodsFedora/rigs-of-rods
[root@vluongo-fedora-hpo rigs-of-rods]# make -j8
Scanning dependencies of target generate_version
Scanning dependencies of target angelscript_addons
[  1%] Generate files containing version information
[  1%] Building CXX object source/angelscript_addons/CMakeFiles/angelscript_addons.dir/scriptany/scriptany.cpp.o
[  1%] Building CXX object source/angelscript_addons/CMakeFiles/angelscript_addons.dir/scriptbuilder/scriptbuilder.cpp.o
[  1%] Building CXX object source/angelscript_addons/CMakeFiles/angelscript_addons.dir/scripthelper/scripthelper.cpp.o
[  2%] Building CXX object source/angelscript_addons/CMakeFiles/angelscript_addons.dir/scriptarray/scriptarray.cpp.o
true
[  3%] Building CXX object source/angelscript_addons/CMakeFiles/angelscript_addons.dir/scriptstdstring/scriptstdstring.cpp.o
[  3%] Building CXX object source/angelscript_addons/CMakeFiles/angelscript_addons.dir/scriptdictionary/scriptdictionary.cpp.o
[  4%] Building CXX object source/angelscript_addons/CMakeFiles/angelscript_addons.dir/scriptmath/scriptmath.cpp.o
[  5%] Building CXX object source/angelscript_addons/CMakeFiles/angelscript_addons.dir/scriptstdstring/scriptstdstring_utils.cpp.o
true
[  5%] Linking CXX static library ../../bin/libangelscript_addons.a
[  5%] Built target angelscript_addons
[  5%] Built target generate_version
Scanning dependencies of target version_info
[  5%] Building CXX object source/version_info/CMakeFiles/version_info.dir/version_info.cpp.o
[  6%] Linking CXX static library ../../bin/libversion_info.a
[  6%] Built target version_info
Scanning dependencies of target RoRConfig
Scanning dependencies of target RoR
[  7%] Building CXX object source/configurator/CMakeFiles/RoRConfig.dir/Configurator.cpp.o
[  7%] Building CXX object source/configurator/CMakeFiles/RoRConfig.dir/SystemTools.cpp.o
[  8%] Building CXX object source/configurator/CMakeFiles/RoRConfig.dir/conf_file.cpp.o
[  8%] Building CXX object source/configurator/CMakeFiles/RoRConfig.dir/statpict.cpp.o
[  9%] Building CXX object source/configurator/CMakeFiles/RoRConfig.dir/__/main/utils/SHA1.cpp.o
[  9%] Building CXX object source/configurator/CMakeFiles/RoRConfig.dir/__/main/utils/ErrorUtils.cpp.o
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/configurator/SystemTools.cpp: In function ‘bool ExtractZipFiles(const wxString&, const wxString&)’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/configurator/SystemTools.cpp:148:49: error: ‘wxPosixPermissions’ has not been declared
             if (! wxFileName::Mkdir(subdirPath, wxPosixPermissions::wxS_DIR_DEFAULT, wxPATH_MKDIR_FULL))
                                                 ^~~~~~~~~~~~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/configurator/SystemTools.cpp:156:70: error: invalid conversion from ‘const char*’ to ‘wxChar {aka wchar_t}’ [-fpermissive]
         if ((lastSepPos != -1) && (entryName.substr(lastSepPos+1) == "empty") && (entry->GetSize() == 0))
                                                                      ^~~~~~~
In file included from /usr/include/wx-2.8/wx/memory.h:16:0,
                 from /usr/include/wx-2.8/wx/object.h:20,
                 from /usr/include/wx-2.8/wx/event.h:17,
                 from /usr/include/wx-2.8/wx/app.h:20,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/configurator/RoRConfig.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/configurator/SystemTools.cpp:27:
/usr/include/wx-2.8/wx/string.h:1634:13: note:   initializing argument 2 of ‘bool operator==(const wxString&, wxChar)’
 inline bool operator==(const wxString& s, wxChar c) { return s.IsSameAs(c); }
             ^~~~~~~~
make[2]: *** [source/configurator/CMakeFiles/RoRConfig.dir/build.make:87: source/configurator/CMakeFiles/RoRConfig.dir/SystemTools.cpp.o] Error 1
make[2]: *** Attesa per i processi non terminati....
[  9%] Building CXX object source/main/CMakeFiles/RoR.dir/main.cpp.o
[  9%] Building CXX object source/main/CMakeFiles/RoR.dir/Application.cpp.o
[ 10%] Building CXX object source/main/CMakeFiles/RoR.dir/MainMenu.cpp.o
[ 10%] Building CXX object source/main/CMakeFiles/RoR.dir/audio/MumbleIntegration.cpp.o
[ 11%] Building CXX object source/main/CMakeFiles/RoR.dir/audio/Sound.cpp.o
[ 11%] Building CXX object source/main/CMakeFiles/RoR.dir/audio/SoundManager.cpp.o
[ 12%] Building CXX object source/main/CMakeFiles/RoR.dir/audio/SoundScriptManager.cpp.o
[ 12%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/AircraftSimulation.cpp.o
[ 13%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/AutoPilot.cpp.o
make[1]: *** [CMakeFiles/Makefile2:312: source/configurator/CMakeFiles/RoRConfig.dir/all] Error 2
make[1]: *** Attesa per i processi non terminati....
[ 13%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/BeamEngine.cpp.o
[ 14%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/Character.cpp.o
[ 14%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/CharacterFactory.cpp.o
[ 15%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/ChatSystem.cpp.o
[ 15%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/Landusemap.cpp.o
[ 16%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/LandVehicleSimulation.cpp.o
[ 16%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/OutProtocol.cpp.o
[ 17%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/PositionStorage.cpp.o
[ 17%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/ProceduralManager.cpp.o
[ 18%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/Replay.cpp.o
[ 18%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/Road.cpp.o
[ 19%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/Road2.cpp.o
[ 19%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/RoRFrameListener.cpp.o
[ 20%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/SceneMouse.cpp.o
[ 20%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/SkinManager.cpp.o
[ 21%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/TorqueCurve.cpp.o
[ 21%] Building CXX object source/main/CMakeFiles/RoR.dir/gameplay/VehicleAI.cpp.o
[ 21%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/ColoredTextAreaOverlayElement.cpp.o
[ 22%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/DecalManager.cpp.o
[ 22%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/DustManager.cpp.o
[ 23%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/DustPool.cpp.o
[ 23%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/EnvironmentMap.cpp.o
[ 24%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/GfxActor.cpp.o
[ 24%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/HDRListener.cpp.o
[ 25%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/Heathaze.cpp.o
[ 25%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/HydraxWater.cpp.o
[ 26%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/MovableText.cpp.o
[ 26%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/OgreSubsystem.cpp.o
[ 27%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/ShadowManager.cpp.o
[ 27%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/Skidmark.cpp.o
[ 28%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/SkyManager.cpp.o
[ 28%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/Water.cpp.o
[ 29%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraBehaviorCharacter.cpp.o
[ 29%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraBehaviorFixed.cpp.o
[ 30%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraBehaviorFree.cpp.o
[ 30%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraBehaviorIsometric.cpp.o
[ 31%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraBehaviorOrbit.cpp.o
[ 31%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraBehaviorStatic.cpp.o
[ 32%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraBehaviorVehicle.cpp.o
[ 32%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraBehaviorVehicleCineCam.cpp.o
[ 33%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraBehaviorVehicleSpline.cpp.o
[ 33%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/camera/CameraManager.cpp.o
[ 34%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/dof/DepthOfFieldEffect.cpp.o
[ 34%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/dof/Lens.cpp.o
[ 34%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/CfgFileManager.cpp.o
[ 35%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/DecalsManager.cpp.o
[ 35%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Enums.cpp.o
[ 36%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/FFT.cpp.o
[ 36%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/GodRaysManager.cpp.o
[ 37%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/GPUNormalMapManager.cpp.o
[ 37%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Help.cpp.o
[ 38%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Hydrax.cpp.o
[ 38%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Image.cpp.o
[ 39%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/MaterialManager.cpp.o
[ 39%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Mesh.cpp.o
[ 40%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Module.cpp.o
[ 40%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Noise.cpp.o
[ 41%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Perlin.cpp.o
[ 41%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Prerequisites.cpp.o
[ 42%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/PressurePoint.cpp.o
[ 42%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/ProjectedGrid.cpp.o
[ 43%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/RadialGrid.cpp.o
[ 43%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Real.cpp.o
[ 44%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/RttManager.cpp.o
[ 44%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/SimpleGrid.cpp.o
[ 45%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/TextureManager.cpp.o
[ 45%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/hydrax/Wave.cpp.o
[ 46%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/particle/ExtinguishableFireAffector.cpp.o
[ 46%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/particle/FireExtinguisherAffector.cpp.o
[ 46%] Building CXX object source/main/CMakeFiles/RoR.dir/gfx/particle/OgreShaderParticleRenderer.cpp.o
[ 47%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/Dashboard.cpp.o
[ 47%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/DashBoardManager.cpp.o
[ 48%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/GUIInputManager.cpp.o
[ 48%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/GUIManager.cpp.o
[ 49%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/GUIUtils.cpp.o
[ 49%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/OverlayWrapper.cpp.o
[ 50%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/RTTLayer.cpp.o
[ 50%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/imgui/imgui.cpp.o
[ 51%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/imgui/imgui_demo.cpp.o
[ 51%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/imgui/imgui_draw.cpp.o
[ 52%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/imgui/OgreImGui.cpp.o
[ 52%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/mygui/Dialog.cpp.o
[ 53%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GameConsole.cpp.o
[ 53%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_LoadingWindow.cpp.o
[ 54%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_FrictionSettings.cpp.o
[ 54%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_TopMenubar.cpp.o
[ 55%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_TextureToolWindow.cpp.o
[ 55%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_DebugOptions.cpp.o
[ 56%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_DebugOptionsLayout.cpp.o
[ 56%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GameAbout.cpp.o
[ 57%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GameAboutLayout.cpp.o
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameConsole.cpp: In constructor ‘RoR::Console::Console()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameConsole.cpp:55:98: warning: ‘void MyGUI::EventPair<EventObsolete, Event>::operator+=(T*) [with T = MyGUI::delegates::IDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; EventObsolete = MyGUI::delegates::CMultiDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; Event = MyGUI::delegates::CMultiDelegate2<MyGUI::Window*, const std::__cxx11::basic_string<char>&>]’ is deprecated [-Wdeprecated-declarations]
     win->eventWindowButtonPressed += MyGUI::newDelegate(this, &Console::notifyWindowButtonPressed); //The "X" button thing
                                                                                                  ^
In file included from /usr/local/include/MYGUI/MyGUI_WidgetInput.h:18:0,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:14,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameConsole.h:33,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameConsole.cpp:23:
/usr/local/include/MYGUI/MyGUI_EventPair.h:37:8: note: declared here
   void operator += (T* _delegate)
        ^~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_DebugOptions.cpp: In constructor ‘RoR::GUI::DebugOptions::DebugOptions()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_DebugOptions.cpp:54:96: warning: ‘void MyGUI::EventPair<EventObsolete, Event>::operator+=(T*) [with T = MyGUI::delegates::IDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; EventObsolete = MyGUI::delegates::CMultiDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; Event = MyGUI::delegates::CMultiDelegate2<MyGUI::Window*, const std::__cxx11::basic_string<char>&>]’ is deprecated [-Wdeprecated-declarations]
     win->eventWindowButtonPressed += MyGUI::newDelegate(this, &CLASS::notifyWindowButtonPressed); //The "X" button thing
                                                                                                ^
In file included from /usr/local/include/MYGUI/MyGUI_WidgetInput.h:18:0,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:14,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_DebugOptionsLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_DebugOptions.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_DebugOptions.cpp:33:
/usr/local/include/MYGUI/MyGUI_EventPair.h:37:8: note: declared here
   void operator += (T* _delegate)
        ^~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAbout.cpp: In constructor ‘RoR::GUI::GameAbout::GameAbout()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAbout.cpp:48:96: warning: ‘void MyGUI::EventPair<EventObsolete, Event>::operator+=(T*) [with T = MyGUI::delegates::IDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; EventObsolete = MyGUI::delegates::CMultiDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; Event = MyGUI::delegates::CMultiDelegate2<MyGUI::Window*, const std::__cxx11::basic_string<char>&>]’ is deprecated [-Wdeprecated-declarations]
     win->eventWindowButtonPressed += MyGUI::newDelegate(this, &CLASS::notifyWindowButtonPressed); //The "X" button thing
                                                                                                ^
In file included from /usr/local/include/MYGUI/MyGUI_WidgetInput.h:18:0,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:14,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAboutLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAbout.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAbout.cpp:26:
/usr/local/include/MYGUI/MyGUI_EventPair.h:37:8: note: declared here
   void operator += (T* _delegate)
        ^~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAbout.cpp: In member function ‘bool RoR::GUI::GameAbout::IsVisible()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAbout.cpp:89:35: warning: ‘bool MyGUI::MemberObsolete<MyGUI::Widget>::isVisible()’ is deprecated [-Wdeprecated-declarations]
     return MAIN_WIDGET->isVisible();
                                   ^
In file included from /usr/local/include/MYGUI/MyGUI_ResourceManager.h:16:0,
                 from /usr/local/include/MYGUI/MyGUI_LayerManager.h:15,
                 from /usr/local/include/MYGUI/MyGUI_LayerItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_SkinItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:18,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAboutLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAbout.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameAbout.cpp:26:
/usr/local/include/MYGUI/MyGUI_BackwardCompatibility.h:377:8: note: declared here
   bool isVisible();
        ^~~~~~~~~
[ 57%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GameChatBox.cpp.o
[ 58%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GameChatBoxLayout.cpp.o
[ 58%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GameConsoleLayout.cpp.o
[ 59%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GameMainMenu.cpp.o
[ 59%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GameMainMenuLayout.cpp.o
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TextureToolWindow.cpp: In constructor ‘RoR::GUI::TextureToolWindow::TextureToolWindow()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TextureToolWindow.cpp:52:94: warning: ‘void MyGUI::EventPair<EventObsolete, Event>::operator+=(T*) [with T = MyGUI::delegates::IDelegate1<MyGUI::Widget*>; EventObsolete = MyGUI::delegates::CMultiDelegate1<MyGUI::Widget*>; Event = MyGUI::delegates::CMultiDelegate2<MyGUI::ComboBox*, long unsigned int>]’ is deprecated [-Wdeprecated-declarations]
     mCBo->eventComboAccept += MyGUI::newDelegate(this, &TextureToolWindow::eventSelectTexture);
                                                                                              ^
In file included from /usr/local/include/MYGUI/MyGUI_WidgetInput.h:18:0,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:14,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TextureToolWindow.h:26,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TextureToolWindow.cpp:22:
/usr/local/include/MYGUI/MyGUI_EventPair.h:37:8: note: declared here
   void operator += (T* _delegate)
        ^~~~~~~~
[ 59%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GamePauseMenu.cpp.o
[ 60%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GamePauseMenuLayout.cpp.o
[ 60%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_GameSettings.cpp.o
[ 61%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_MainSelector.cpp.o
[ 61%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_MainSelectorLayout.cpp.o
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameChatBox.cpp: In member function ‘bool RoR::GUI::GameChatBox::IsVisible()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameChatBox.cpp:85:35: warning: ‘bool MyGUI::MemberObsolete<MyGUI::Widget>::isVisible()’ is deprecated [-Wdeprecated-declarations]
     return MAIN_WIDGET->isVisible();
                                   ^
In file included from /usr/local/include/MYGUI/MyGUI_ResourceManager.h:16:0,
                 from /usr/local/include/MYGUI/MyGUI_LayerManager.h:15,
                 from /usr/local/include/MYGUI/MyGUI_LayerItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_SkinItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:18,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameChatBoxLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameChatBox.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_GameChatBox.cpp:26:
/usr/local/include/MYGUI/MyGUI_BackwardCompatibility.h:377:8: note: declared here
   bool isVisible();
        ^~~~~~~~~
[ 62%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_MessageBox.cpp.o
[ 62%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_MessageBoxLayout.cpp.o
[ 63%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_MultiplayerSelector.cpp.o
[ 63%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_MultiplayerClientList.cpp.o
[ 64%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_OpenSaveFileDialog.cpp.o
[ 64%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_RigSpawnerReportWindow.cpp.o
[ 65%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_RigSpawnerReportWindowLayout.cpp.o
[ 65%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_SimUtils.cpp.o
[ 66%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_SimUtilsLayout.cpp.o
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.cpp: In constructor ‘RoR::GUI::MainSelector::MainSelector()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.cpp:55:96: warning: ‘void MyGUI::EventPair<EventObsolete, Event>::operator+=(T*) [with T = MyGUI::delegates::IDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; EventObsolete = MyGUI::delegates::CMultiDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; Event = MyGUI::delegates::CMultiDelegate2<MyGUI::Window*, const std::__cxx11::basic_string<char>&>]’ is deprecated [-Wdeprecated-declarations]
     win->eventWindowButtonPressed += MyGUI::newDelegate(this, &CLASS::NotifyWindowButtonPressed); //The "X" button thing
                                                                                                ^
In file included from /usr/local/include/MYGUI/MyGUI_WidgetInput.h:18:0,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:14,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelectorLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.cpp:22:
/usr/local/include/MYGUI/MyGUI_EventPair.h:37:8: note: declared here
   void operator += (T* _delegate)
        ^~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.cpp: In member function ‘void RoR::GUI::MainSelector::NotifyWindowChangeCoord(MyGUI::Window*)’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.cpp:122:30: warning: ‘bool MyGUI::MemberObsolete<MyGUI::Widget>::isVisible()’ is deprecated [-Wdeprecated-declarations]
     if (m_Preview->isVisible())
                              ^
In file included from /usr/local/include/MYGUI/MyGUI_ResourceManager.h:16:0,
                 from /usr/local/include/MYGUI/MyGUI_LayerManager.h:15,
                 from /usr/local/include/MYGUI/MyGUI_LayerItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_SkinItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:18,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelectorLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.cpp:22:
/usr/local/include/MYGUI/MyGUI_BackwardCompatibility.h:377:8: note: declared here
   bool isVisible();
        ^~~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.cpp: In member function ‘bool RoR::GUI::MainSelector::IsVisible()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.cpp:1057:35: warning: ‘bool MyGUI::MemberObsolete<MyGUI::Widget>::isVisible()’ is deprecated [-Wdeprecated-declarations]
     return MAIN_WIDGET->isVisible();
                                   ^
In file included from /usr/local/include/MYGUI/MyGUI_ResourceManager.h:16:0,
                 from /usr/local/include/MYGUI/MyGUI_LayerManager.h:15,
                 from /usr/local/include/MYGUI/MyGUI_LayerItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_SkinItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:18,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelectorLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MainSelector.cpp:22:
/usr/local/include/MYGUI/MyGUI_BackwardCompatibility.h:377:8: note: declared here
   bool isVisible();
        ^~~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MessageBox.cpp: In constructor ‘RoR::GUI::gMessageBox::gMessageBox()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MessageBox.cpp:48:96: warning: ‘void MyGUI::EventPair<EventObsolete, Event>::operator+=(T*) [with T = MyGUI::delegates::IDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; EventObsolete = MyGUI::delegates::CMultiDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; Event = MyGUI::delegates::CMultiDelegate2<MyGUI::Window*, const std::__cxx11::basic_string<char>&>]’ is deprecated [-Wdeprecated-declarations]
     win->eventWindowButtonPressed += MyGUI::newDelegate(this, &CLASS::notifyWindowButtonPressed); //The "X" button thing
                                                                                                ^
In file included from /usr/local/include/MYGUI/MyGUI_WidgetInput.h:18:0,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:14,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MessageBoxLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MessageBox.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_MessageBox.cpp:26:
/usr/local/include/MYGUI/MyGUI_EventPair.h:37:8: note: declared here
   void operator += (T* _delegate)
        ^~~~~~~~
[ 66%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_TeleportWindow.cpp.o
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_RigSpawnerReportWindow.cpp: In constructor ‘RoR::GUI::RigSpawnerReportWindow::RigSpawnerReportWindow()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_RigSpawnerReportWindow.cpp:36:115: warning: ‘void MyGUI::EventPair<EventObsolete, Event>::operator+=(T*) [with T = MyGUI::delegates::IDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; EventObsolete = MyGUI::delegates::CMultiDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; Event = MyGUI::delegates::CMultiDelegate2<MyGUI::Window*, const std::__cxx11::basic_string<char>&>]’ is deprecated [-Wdeprecated-declarations]
     main_window->eventWindowButtonPressed += MyGUI::newDelegate(this, &RigSpawnerReportWindow::WindowButtonClicked);
                                                                                                                   ^
In file included from /usr/local/include/MYGUI/MyGUI_WidgetInput.h:18:0,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:14,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_RigSpawnerReportWindowLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_RigSpawnerReportWindow.h:29,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_RigSpawnerReportWindow.cpp:26:
/usr/local/include/MYGUI/MyGUI_EventPair.h:37:8: note: declared here
   void operator += (T* _delegate)
        ^~~~~~~~
[ 67%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_TeleportWindowLayout.cpp.o
[ 67%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_VehicleDescription.cpp.o
[ 68%] Building CXX object source/main/CMakeFiles/RoR.dir/gui/panels/GUI_VehicleDescriptionLayout.cpp.o
[ 68%] Building CXX object source/main/CMakeFiles/RoR.dir/network/Network.cpp.o
[ 69%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/Beam.cpp.o
[ 69%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/BeamFactory.cpp.o
[ 70%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/BeamForcesEuler.cpp.o
[ 70%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/BeamSlideNode.cpp.o
[ 71%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/CmdKeyInertia.cpp.o
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.cpp: In constructor ‘RoR::GUI::TeleportWindow::TeleportWindow()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.cpp:46:113: warning: ‘void MyGUI::EventPair<EventObsolete, Event>::operator+=(T*) [with T = MyGUI::delegates::IDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; EventObsolete = MyGUI::delegates::CMultiDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; Event = MyGUI::delegates::CMultiDelegate2<MyGUI::Window*, const std::__cxx11::basic_string<char>&>]’ is deprecated [-Wdeprecated-declarations]
     m_teleport_window->eventWindowButtonPressed += MyGUI::newDelegate(this, &TeleportWindow::WindowButtonClicked);
                                                                                                                 ^
In file included from /usr/local/include/MYGUI/MyGUI_WidgetInput.h:18:0,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:14,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindowLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.h:23,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.cpp:20:
/usr/local/include/MYGUI/MyGUI_EventPair.h:37:8: note: declared here
   void operator += (T* _delegate)
        ^~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.cpp: In member function ‘void RoR::GUI::TeleportWindow::SetVisible(bool)’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.cpp:79:38: warning: ‘bool MyGUI::MemberObsolete<MyGUI::Widget>::isVisible()’ is deprecated [-Wdeprecated-declarations]
     if (m_teleport_window->isVisible() == v)
                                      ^
In file included from /usr/local/include/MYGUI/MyGUI_ResourceManager.h:16:0,
                 from /usr/local/include/MYGUI/MyGUI_LayerManager.h:15,
                 from /usr/local/include/MYGUI/MyGUI_LayerItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_SkinItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:18,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindowLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.h:23,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.cpp:20:
/usr/local/include/MYGUI/MyGUI_BackwardCompatibility.h:377:8: note: declared here
   bool isVisible();
        ^~~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.cpp: In member function ‘void RoR::GUI::TeleportWindow::MinimapMouseMoved(MyGUI::Widget*, int, int)’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.cpp:142:33: warning: ‘bool MyGUI::MemberObsolete<MyGUI::Widget>::isVisible()’ is deprecated [-Wdeprecated-declarations]
     if (m_mouse_icon->isVisible())
                                 ^
In file included from /usr/local/include/MYGUI/MyGUI_ResourceManager.h:16:0,
                 from /usr/local/include/MYGUI/MyGUI_LayerManager.h:15,
                 from /usr/local/include/MYGUI/MyGUI_LayerItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_SkinItem.h:11,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:18,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindowLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.h:23,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_TeleportWindow.cpp:20:
/usr/local/include/MYGUI/MyGUI_BackwardCompatibility.h:377:8: note: declared here
   bool isVisible();
        ^~~~~~~~~
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_VehicleDescription.cpp: In constructor ‘RoR::GUI::VehicleDescription::VehicleDescription()’:
/home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_VehicleDescription.cpp:50:96: warning: ‘void MyGUI::EventPair<EventObsolete, Event>::operator+=(T*) [with T = MyGUI::delegates::IDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; EventObsolete = MyGUI::delegates::CMultiDelegate2<MyGUI::Widget*, const std::__cxx11::basic_string<char>&>; Event = MyGUI::delegates::CMultiDelegate2<MyGUI::Window*, const std::__cxx11::basic_string<char>&>]’ is deprecated [-Wdeprecated-declarations]
     win->eventWindowButtonPressed += MyGUI::newDelegate(this, &CLASS::notifyWindowButtonPressed); //The "X" button thing
                                                                                                ^
In file included from /usr/local/include/MYGUI/MyGUI_WidgetInput.h:18:0,
                 from /usr/local/include/MYGUI/MyGUI_Widget.h:14,
                 from /usr/local/include/MYGUI/MyGUI_Gui.h:15,
                 from /usr/local/include/MYGUI/MyGUI.h:10,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/mygui/BaseLayout.h:28,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_VehicleDescriptionLayout.h:12,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_VehicleDescription.h:27,
                 from /home/vluongo/RigsOfRodsFedora/rigs-of-rods/source/main/gui/panels/GUI_VehicleDescription.cpp:26:
/usr/local/include/MYGUI/MyGUI_EventPair.h:37:8: note: declared here
   void operator += (T* _delegate)
        ^~~~~~~~
[ 71%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/Differentials.cpp.o
[ 71%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/RigSpawner.cpp.o
[ 72%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/SlideNode.cpp.o
[ 72%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/RigSpawner_ProcessControl.cpp.o
[ 73%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/air/AirBrake.cpp.o
[ 73%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/air/Airfoil.cpp.o
[ 74%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/air/TurboJet.cpp.o
[ 74%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/air/TurboProp.cpp.o
[ 75%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/collision/Collisions.cpp.o
[ 75%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/collision/DynamicCollisions.cpp.o
[ 76%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/collision/PointColDetector.cpp.o
[ 76%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/flex/FlexAirfoil.cpp.o
[ 77%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/flex/FlexBody.cpp.o
[ 77%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/flex/FlexFactory.cpp.o
[ 78%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/flex/FlexMesh.cpp.o
[ 78%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/flex/FlexMeshWheel.cpp.o
[ 79%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/flex/FlexObj.cpp.o
[ 79%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/mplatform/MPlatformBase.cpp.o
[ 80%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/mplatform/MPlatformFD.cpp.o
[ 80%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/utils/BeamStats.cpp.o
[ 81%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/water/Buoyance.cpp.o
[ 81%] Building CXX object source/main/CMakeFiles/RoR.dir/physics/water/ScrewProp.cpp.o
[ 82%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/CacheSystem.cpp.o
[ 82%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/ContentManager.cpp.o
[ 83%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/otc_fileformat/OTCFileformat.cpp.o
[ 83%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/rig_def_fileformat/RigDef_File.cpp.o
[ 84%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/rig_def_fileformat/RigDef_Node.cpp.o
[ 84%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/rig_def_fileformat/RigDef_Parser.cpp.o
[ 84%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/rig_def_fileformat/RigDef_SequentialImporter.cpp.o
[ 85%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/rig_def_fileformat/RigDef_Serializer.cpp.o
[ 85%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/rig_def_fileformat/RigDef_Validator.cpp.o
[ 86%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/terrn2_fileformat/Terrn2Fileformat.cpp.o
[ 86%] Building CXX object source/main/CMakeFiles/RoR.dir/terrain/OgreTerrainPSSMMaterialGenerator.cpp.o
[ 87%] Building CXX object source/main/CMakeFiles/RoR.dir/terrain/TerrainGeometryManager.cpp.o
[ 87%] Building CXX object source/main/CMakeFiles/RoR.dir/terrain/TerrainManager.cpp.o
[ 88%] Building CXX object source/main/CMakeFiles/RoR.dir/terrain/TerrainObjectManager.cpp.o
[ 88%] Building CXX object source/main/CMakeFiles/RoR.dir/terrain/map/SurveyMapEntity.cpp.o
[ 89%] Building CXX object source/main/CMakeFiles/RoR.dir/terrain/map/SurveyMapManager.cpp.o
[ 89%] Building CXX object source/main/CMakeFiles/RoR.dir/terrain/map/SurveyMapTextureCreator.cpp.o
[ 90%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/CollisionTools.cpp.o
[ 90%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/ConfigFile.cpp.o
[ 91%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/ErrorUtils.cpp.o
[ 91%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/ForceFeedback.cpp.o
[ 92%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/InputEngine.cpp.o
[ 92%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/Language.cpp.o
[ 93%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/MeshObject.cpp.o
[ 93%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/PlatformUtils.cpp.o
[ 94%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/RoRWindowEventUtilities.cpp.o
[ 94%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/Settings.cpp.o
[ 95%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/SHA1.cpp.o
[ 95%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/Utils.cpp.o
[ 96%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/WriteTextToTexture.cpp.o
[ 96%] Building CXX object source/main/CMakeFiles/RoR.dir/utils/profiler/Profiler.cpp.o
[ 96%] Building CXX object source/main/CMakeFiles/RoR.dir/scripting/CBytecodeStream.cpp.o
[ 97%] Building CXX object source/main/CMakeFiles/RoR.dir/scripting/GameScript.cpp.o
[ 97%] Building CXX object source/main/CMakeFiles/RoR.dir/scripting/LocalStorage.cpp.o
[ 98%] Building CXX object source/main/CMakeFiles/RoR.dir/scripting/OgreAngelscript.cpp.o
[ 98%] Building CXX object source/main/CMakeFiles/RoR.dir/scripting/OgreScriptBuilder.cpp.o
[ 99%] Building CXX object source/main/CMakeFiles/RoR.dir/scripting/ScriptEngine.cpp.o
[ 99%] Linking CXX executable ../../bin/RoR
Zip and copy resources to build directory
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/textures.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/scripts.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/glow.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/caelum.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/beamobjects.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/overlays.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/meshes.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/rtshader.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/famicons.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/wallpapers.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/hydrax.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/dashboards.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/icons.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/particles.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/sunburn.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/paged.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/heathaze.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/blur.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/OgreCore.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/sounds.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/skeleton.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/hdr.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/airfoils.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/dof.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/cubemaps.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/materials.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/managed_materials.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/managed_materials.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/managed_materials.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/managed_materials.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/managed_materials.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/managed_materials.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/flags.zip
Created /home/vluongo/RigsOfRodsFedora/rigs-of-rods/bin/resources/mygui.zip
[ 99%] Built target RoR
make: *** [Makefile:152: all] Error 2
```

last version not build.
```
rigs-of-rods/source/configurator/SystemTools.cpp:148:49: error: ‘wxPosixPermissions’ has not been declared
```

but I have create simple way for build old version.
wait fix next release

@only-a-ptr if I try to build with wx3.0 not have this error but have another error:
```
In file included from /usr/include/gtk-2.0/gdk/gdkscreen.h:32:0,
                 from /usr/include/gtk-2.0/gdk/gdkapplaunchcontext.h:31,
                 from /usr/include/gtk-2.0/gdk/gdk.h:32,
                 from /usr/include/gtk-2.0/gtk/gtk.h:32,
                 from /home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/wxutils.h:7,
                 from /home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/Configurator.cpp:79:
/usr/include/gtk-2.0/gdk/gdktypes.h:114:39: error: conflicting declaration ‘typedef struct _GdkDrawable GdkWindow’
 typedef struct _GdkDrawable           GdkWindow;
                                       ^~~~~~~~~
In file included from /usr/include/wx-3.0/wx/event.h:14:0,
                 from /usr/include/wx-3.0/wx/app.h:19,
                 from /home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/RoRConfig.h:29,
                 from /home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/Configurator.cpp:22:
/usr/include/wx-3.0/wx/defs.h:3464:31: note: previous declaration as ‘typedef struct _GdkWindow GdkWindow’
     typedef struct _GdkWindow GdkWindow;
                               ^~~~~~~~~
In file included from /home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/wxutils.h:8:0,
                 from /home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/Configurator.cpp:79:
/home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/wxutils.h: In function ‘size_t getOISHandle(wxWindow*)’:
/usr/include/gtk-2.0/gdk/gdkx.h:113:69: error: cannot convert ‘GdkWindow* {aka _GdkWindow*}’ to ‘GdkDrawable* {aka _GdkDrawable*}’ for argument ‘1’ to ‘XID gdk_x11_drawable_get_xid(GdkDrawable*)’
 #define GDK_WINDOW_XID(win)           (gdk_x11_drawable_get_xid (win))
                                                                     ^
/home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/wxutils.h:54:20: note: in expansion of macro ‘GDK_WINDOW_XID’
     hWnd = (size_t)GDK_WINDOW_XID(gtk_widget_get_window(window->GetHandle()));
                    ^
In file included from /usr/include/wx-3.0/wx/event.h:14:0,
                 from /usr/include/wx-3.0/wx/app.h:19,
                 from /home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/RoRConfig.h:29,
                 from /home/vbextreme/Sources/rigsofrods/rigs-of-rods/source/configurator/Configurator.cpp:22:
/usr/include/wx-3.0/wx/defs.h:3464:20: note: class type ‘GdkWindow {aka _GdkWindow}’ is incomplete
     typedef struct _GdkWindow GdkWindow;
```

--- a/source/configurator/SystemTools.cpp
+++ b/source/configurator/SystemTools.cpp
@@ -144,8 +144,9 @@ bool ExtractZipFiles(const wxString& aZipFile, const wxString& aTargetDir)
         if (lastSepPos != -1)
         {
             wxString subdirPath = aTargetDir + wxFileName::GetPathSeparator() + entryName.SubString(0, lastSepPos);
+            const int permissions = 0777; // Equivalent of 'wxPosixPermissions::wxS_DIR_DEFAULT' in WxWidgets 3.x, we want to remain 2.x compatible.
             // With flag 'wxPATH_MKDIR_FULL', existing directories are OK
-            if (! wxFileName::Mkdir(subdirPath, wxPosixPermissions::wxS_DIR_DEFAULT, wxPATH_MKDIR_FULL))
+            if (! wxFileName::Mkdir(subdirPath, permissions, wxPATH_MKDIR_FULL))
             {
                 wxLogError(_T("Could not create full subdirectory path '") + subdirPath + wxT("'."));
                 continue;
@@ -153,7 +154,7 @@ bool ExtractZipFiles(const wxString& aZipFile, const wxString& aTargetDir)
         }
 
         // Skip dummy 'keep empty directory' files
-        if ((lastSepPos != -1) && (entryName.substr(lastSepPos+1) == "empty") && (entry->GetSize() == 0))
+        if ((lastSepPos != -1) && (entryName.substr(lastSepPos+1) == wxString("empty")) && (entry->GetSize() == 0))
         {
             continue;
         }

--- a/source/main/Application.cpp
+++ b/source/main/Application.cpp
@@ -34,6 +34,7 @@
 #include "OgreSubsystem.h"
 #include "OverlayWrapper.h"
 #include "SceneMouse.h"
+#include "MumbleIntegration.h"
 
 namespace RoR {
 namespace App {
@@ -54,6 +55,7 @@ static InputEngine*     g_input_engine;
 static CacheSystem*     g_cache_system;
 static MainMenu*        g_main_menu;
 static RoRFrameListener* g_sim_controller;
+static MumbleIntegration* g_mumble;
 
 // App
  GVarEnum_AP<AppState>    app_state               ("app_state",               nullptr,                     AppState::BOOTSTRAP,     AppState::MAIN_MENU);
@@ -174,6 +176,7 @@ InputEngine*           GetInputEngine        () { return g_input_engine;}
 CacheSystem*           GetCacheSystem        () { return g_cache_system;}
 MainMenu*              GetMainMenu           () { return g_main_menu;}
 RoRFrameListener*      GetSimController      () { return g_sim_controller; }
+MumbleIntegration*     GetMumble             () { return g_mumble; }
 
 void StartOgreSubsystem()
 {
@@ -264,6 +267,15 @@ void CreateCacheSystem()
 {
     assert(g_cache_system == nullptr);
     g_cache_system = new CacheSystem();
+}
+
+
+void CheckAndCreateMumble()
+{
+#ifdef USE_MUMBLE // The class is always forward-declared but only defined if USE_MUMBLE is defined
+    if (g_mumble == nullptr)
+        g_mumble = new MumbleIntegration();
+#endif // USE_MUMBLE
 }
 
 } // namespace App

--- a/source/main/Application.h
+++ b/source/main/Application.h
@@ -763,6 +763,7 @@ InputEngine*         GetInputEngine();
 CacheSystem*         GetCacheSystem();
 MainMenu*            GetMainMenu();
 RoRFrameListener*    GetSimController();
+MumbleIntegration*   GetMumble();
 
 // Factories
 void StartOgreSubsystem();
@@ -777,6 +778,7 @@ void CreateGuiManagerIfNotExists();
 void DeleteGuiManagerIfExists();
 void CreateInputEngine();
 void CreateCacheSystem();
+void CheckAndCreateMumble();
 
 // Setters
 void SetMainMenu             (MainMenu*          obj);

--- a/source/main/Application.h
+++ b/source/main/Application.h
@@ -764,8 +764,6 @@ CacheSystem*         GetCacheSystem();
 MainMenu*            GetMainMenu();
 RoRFrameListener*    GetSimController();
 
-void SetMainMenu             (MainMenu*          obj);
-void SetSimController        (RoRFrameListener*  obj);
 // Factories
 void StartOgreSubsystem();
 void ShutdownOgreSubsystem();

--- a/source/main/MainMenu.cpp
+++ b/source/main/MainMenu.cpp
@@ -220,9 +220,7 @@ void MainMenu::MainMenuLoopUpdate(float seconds_since_last_frame)
             App::mp_state.SetActive(RoR::MpState::CONNECTED);
             gui->SetVisible_MpClientList(true);
             ChatSystem::SendStreamSetup();
-#ifdef USE_MUMBLE
-            SoundScriptManager::getSingleton().CheckAndCreateMumble();
-#endif // USE_MUMBLE
+            App::CheckAndCreateMumble();
             String terrain_name = Networking::GetTerrainName();
             if (terrain_name != "any")
             {

--- a/source/main/audio/MumbleIntegration.h
+++ b/source/main/audio/MumbleIntegration.h
@@ -23,9 +23,9 @@
 /// @author Thomas Fischer (thomas{AT}thomasfischer{DOT}biz)
 /// @date   26th of March 2011
 
-#ifdef USE_MUMBLE
-
 #pragma once
+
+#ifdef USE_MUMBLE
 
 #include "RoRPrerequisites.h"
 

--- a/source/main/audio/SoundScriptManager.cpp
+++ b/source/main/audio/SoundScriptManager.cpp
@@ -51,7 +51,6 @@ SoundScriptManager::SoundScriptManager() :
     , rolloff_factor(1.0f)
     , reference_distance(7.5f)
     , sound_manager(nullptr)
-    , m_mumble(nullptr)
 {
     for (int i = 0; i < SS_MAX_TRIG; i++)
     {
@@ -1396,13 +1395,5 @@ void SoundScriptInstance::setEnabled(bool e)
         }
     }
 }
-
-#ifdef USE_MUMBLE
-void SoundScriptManager::CheckAndCreateMumble()
-{
-    if (m_mumble == nullptr)
-        m_mumble = new MumbleIntegration();
-}
-#endif // USE_MUMBLE
 
 #endif // USE_OPENAL

--- a/source/main/audio/SoundScriptManager.h
+++ b/source/main/audio/SoundScriptManager.h
@@ -282,11 +282,6 @@ public:
 
     bool isDisabled() { return disabled; }
 
-#ifdef USE_MUMBLE
-    MumbleIntegration* GetMumble() { return m_mumble; }
-    void               CheckAndCreateMumble();
-#endif // USE_MUMBLE
-
 private:
 
     SoundScriptTemplate* createTemplate(Ogre::String name, Ogre::String groupname, Ogre::String filename);
@@ -318,9 +313,6 @@ private:
     std::map <int, std::map <int, std::map <int, std::map <int, bool > > > > state_map;
 
     SoundManager* sound_manager;
-#ifdef USE_MUMBLE
-    MumbleIntegration* m_mumble; // TODO: Not a perfect spot, but better than `class MainMenu`.
-#endif // USE_MUMBLE
 };
 
 #endif // USE_OPENAL

--- a/source/main/audio/SoundScriptManager.h
+++ b/source/main/audio/SoundScriptManager.h
@@ -28,6 +28,14 @@
 
 #include <OgreScriptLoader.h>
 
+#define SOUND_PLAY_ONCE(_ACTOR_, _TRIG_)        SoundScriptManager::getSingleton().trigOnce    (_ACTOR_, _TRIG_)
+#define SOUND_START(_ACTOR_, _TRIG_)            SoundScriptManager::getSingleton().trigStart   (_ACTOR_, _TRIG_)
+#define SOUND_STOP(_ACTOR_, _TRIG_)             SoundScriptManager::getSingleton().trigStop    (_ACTOR_, _TRIG_)
+#define SOUND_TOGGLE(_ACTOR_, _TRIG_)           SoundScriptManager::getSingleton().trigToggle  (_ACTOR_, _TRIG_)
+#define SOUND_KILL(_ACTOR_, _TRIG_)             SoundScriptManager::getSingleton().trigKill    (_ACTOR_, _TRIG_)
+#define SOUND_GET_STATE(_ACTOR_, _TRIG_)        SoundScriptManager::getSingleton().getTrigState(_ACTOR_, _TRIG_)
+#define SOUND_MODULATE(_ACTOR_, _MOD_, _VALUE_) SoundScriptManager::getSingleton().modulate    (_ACTOR_, _MOD_, _VALUE_)
+
 enum {
     MAX_SOUNDS_PER_SCRIPT = 16,
     MAX_INSTANCES_PER_GROUP = 256
@@ -314,5 +322,15 @@ private:
 
     SoundManager* sound_manager;
 };
+
+#else // USE_OPENAL
+
+#define SOUND_PLAY_ONCE(_ACTOR_, _TRIG_)
+#define SOUND_START(_ACTOR_, _TRIG_)
+#define SOUND_STOP(_ACTOR_, _TRIG_)
+#define SOUND_TOGGLE(_ACTOR_, _TRIG_)
+#define SOUND_KILL(_ACTOR_, _TRIG_)
+#define SOUND_GET_STATE(_ACTOR_, _TRIG_) (false)
+#define SOUND_MODULATE(_ACTOR_, _MOD_, _VALUE_)
 
 #endif // USE_OPENAL

--- a/source/main/gameplay/AutoPilot.cpp
+++ b/source/main/gameplay/AutoPilot.cpp
@@ -28,9 +28,9 @@
 
 using namespace Ogre;
 
-Autopilot::Autopilot(int trucknum)
+Autopilot::Autopilot(int actor_id):
+    m_actor_id(actor_id)
 {
-    this->trucknum = trucknum;
     ref_l = nullptr;
     ref_r = nullptr;
     ref_b = nullptr;
@@ -67,9 +67,7 @@ void Autopilot::disconnect()
     wantsdisconnect = false;
     if (mode_gpws)
     {
-#ifdef USE_OPENAL
-        SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_GPWS_APDISCONNECT);
-#endif //OPENAL
+        SOUND_PLAY_ONCE(m_actor_id, SS_TRIG_GPWS_APDISCONNECT);
     }
 }
 
@@ -328,17 +326,17 @@ void Autopilot::gpws_update(float spawnheight)
         if ((ref_c->Velocity.length() * 1.9685f) > 10.0f)
         {
             if (height < 10 && last_gpws_height > 10)
-                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_GPWS_10);
+                SOUND_PLAY_ONCE(m_actor_id, SS_TRIG_GPWS_10);
             if (height < 20 && last_gpws_height > 20)
-                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_GPWS_20);
+                SOUND_PLAY_ONCE(m_actor_id, SS_TRIG_GPWS_20);
             if (height < 30 && last_gpws_height > 30)
-                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_GPWS_30);
+                SOUND_PLAY_ONCE(m_actor_id, SS_TRIG_GPWS_30);
             if (height < 40 && last_gpws_height > 40)
-                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_GPWS_40);
+                SOUND_PLAY_ONCE(m_actor_id, SS_TRIG_GPWS_40);
             if (height < 50 && last_gpws_height > 50)
-                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_GPWS_50);
+                SOUND_PLAY_ONCE(m_actor_id, SS_TRIG_GPWS_50);
             if (height < 100 && last_gpws_height > 100)
-                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_GPWS_100);
+                SOUND_PLAY_ONCE(m_actor_id, SS_TRIG_GPWS_100);
         }
         last_gpws_height = height;
 
@@ -349,7 +347,7 @@ void Autopilot::gpws_update(float spawnheight)
         float yVel = ref_c->Velocity.y * 1.9685f;
         // will trigger the pullup sound when vvi is high (avoid pullup warning when landing normal) and groundcontact will be in less then 10 seconds
         if (yVel * 10.0f < -height && yVel < -10.0f)
-            SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_GPWS_PULLUP);
+            SOUND_PLAY_ONCE(m_actor_id, SS_TRIG_GPWS_PULLUP);
     }
 #endif //OPENAL
 }
@@ -424,10 +422,12 @@ void Autopilot::getRadioFix(TerrainObjectManager::localizer_t* localizers, int f
     lastradiov = closest_vangle;
     *hdev = closest_hangle;
     *vdev = closest_vangle;
-#ifdef USE_OPENAL
+
     if (mode_heading == HEADING_NAV && mode_gpws && closest_hdist > 10.0 && closest_hdist < 350.0 && last_closest_hdist > 10.0 && last_closest_hdist > 350.0)
-        SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_GPWS_MINIMUMS);
-#endif //OPENAL
+    {
+        SOUND_PLAY_ONCE(m_actor_id, SS_TRIG_GPWS_MINIMUMS);
+    }
+
     last_closest_hdist = closest_hdist;
     if (mode_heading == HEADING_NAV && (closest_hdist < 20.0 || closest_vdist < 20.0))
         wantsdisconnect = true;

--- a/source/main/gameplay/AutoPilot.h
+++ b/source/main/gameplay/AutoPilot.h
@@ -100,5 +100,5 @@ private:
     float lastradiorwd;
     float last_closest_hdist;
 
-    int trucknum;
+    int m_actor_id;
 };

--- a/source/main/gameplay/BeamEngine.cpp
+++ b/source/main/gameplay/BeamEngine.cpp
@@ -260,9 +260,7 @@ void BeamEngine::update(float dt, int doUpdate)
 
     if (doUpdate)
     {
-#ifdef USE_OPENAL
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_INJECTOR, acc);
-#endif // USE_OPENAL
+        SOUND_MODULATE(m_actor, SS_MOD_INJECTOR, acc);
     }
 
     if (hasair)
@@ -271,9 +269,7 @@ void BeamEngine::update(float dt, int doUpdate)
         apressure += dt * curEngineRPM;
         if (apressure > 50000.0f)
         {
-#ifdef USE_OPENAL
-            SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_AIR_PURGE);
-#endif // USE_OPENAL
+            SOUND_PLAY_ONCE(m_actor, SS_TRIG_AIR_PURGE);
             apressure = 0.0f;
         }
     }
@@ -336,11 +332,11 @@ void BeamEngine::update(float dt, int doUpdate)
 
                             if (b_flutter)
                             {
-                                SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_TURBOWASTEGATE);
+                                SOUND_START(m_actor, SS_TRIG_TURBOWASTEGATE);
                                 if (curTurboRPM[i] < minWGPsi * wastegate_threshold_n)
                                 {
                                     b_flutter = false;
-                                    SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_TURBOWASTEGATE);
+                                    SOUND_STOP(m_actor, SS_TRIG_TURBOWASTEGATE);
                                 }
                             }
                         }
@@ -380,11 +376,13 @@ void BeamEngine::update(float dt, int doUpdate)
                         if (curTurboRPM[i] > maxTurboRPM * 0.35 && curTurboRPM[i] < maxTurboRPM)
                         {
                             turbotorque -= (turbotorque * (f * antilag_power_factor));
-                            SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_TURBOBACKFIRE);
+                            SOUND_START(m_actor, SS_TRIG_TURBOBACKFIRE);
                         }
                     }
                     else
-                        SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_TURBOBACKFIRE);
+                    {
+                        SOUND_STOP(m_actor, SS_TRIG_TURBOBACKFIRE);
+                    }
                 }
 
                 // update main turbo rpm
@@ -403,12 +401,12 @@ void BeamEngine::update(float dt, int doUpdate)
 
                     if (curAcc < 0.06 && curTurboRPM[i] > minBOVPsi * 10000)
                     {
-                        SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_TURBOBOV);
+                        SOUND_START(m_actor, SS_TRIG_TURBOBOV);
                         curBOVTurboRPM[i] += dt * turboBOVtorque / (turboInertia * 0.1);
                     }
                     else
                     {
-                        SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_TURBOBOV);
+                        SOUND_STOP(m_actor, SS_TRIG_TURBOBOV);
                         if (curBOVTurboRPM[i] < curTurboRPM[i])
                             curBOVTurboRPM[i] += dt * turboBOVtorque / (turboInertia * 0.05);
                         else
@@ -459,9 +457,7 @@ void BeamEngine::update(float dt, int doUpdate)
             else
             {
                 running = true;
-#ifdef USE_OPENAL
-                SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_ENGINE);
-#endif // USE_OPENAL
+                SOUND_START(m_actor, SS_TRIG_ENGINE);
             }
         }
     }
@@ -512,9 +508,7 @@ void BeamEngine::update(float dt, int doUpdate)
                 else
                 {
                     // engage a gear
-#ifdef USE_OPENAL
-                    SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_SHIFT);
-#endif // USE_OPENAL
+                    SOUND_START(m_actor, SS_TRIG_SHIFT);
                     if (autoselect != NEUTRAL)
                     {
                         curGear += shiftval;
@@ -528,9 +522,8 @@ void BeamEngine::update(float dt, int doUpdate)
             if (shiftclock > shift_time)
             {
                 // we're done shifting
-#ifdef USE_OPENAL
-                SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_SHIFT);
-#endif // USE_OPENAL
+                SOUND_STOP(m_actor, SS_TRIG_SHIFT);
+
                 setAcc(autocurAcc);
                 shifting = 0;
                 postshifting = 1;
@@ -811,28 +804,26 @@ void BeamEngine::update(float dt, int doUpdate)
 void BeamEngine::updateAudio(int doUpdate)
 {
 #ifdef USE_OPENAL
-    const int trucknum = m_actor->trucknum;
-
     if (hasturbo)
     {
         for (int i = 0; i < numTurbos; i++)
-            SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_TURBO, curTurboRPM[i]);
+             SOUND_MODULATE(m_actor, SS_MOD_TURBO, curTurboRPM[i]);
     }
 
     if (doUpdate)
     {
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_ENGINE, curEngineRPM);
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_TORQUE, curClutchTorque);
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_GEARBOX, curWheelRevolutions);
+        SOUND_MODULATE(m_actor, SS_MOD_ENGINE, curEngineRPM);
+        SOUND_MODULATE(m_actor, SS_MOD_TORQUE, curClutchTorque);
+        SOUND_MODULATE(m_actor, SS_MOD_GEARBOX, curWheelRevolutions);
     }
     // reverse gear beep
     if (curGear == -1 && running)
     {
-        SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_REVERSE_GEAR);
+        SOUND_START(m_actor, SS_TRIG_REVERSE_GEAR);
     }
     else
     {
-        SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_REVERSE_GEAR);
+        SOUND_STOP(m_actor, SS_TRIG_REVERSE_GEAR);
     }
 #endif // USE_OPENAL
 }
@@ -995,16 +986,14 @@ float BeamEngine::getClutchForce()
 void BeamEngine::toggleContact()
 {
     contact = !contact;
-#ifdef USE_OPENAL
     if (contact)
     {
-        SoundScriptManager::getSingleton().trigStart(m_actor->trucknum, SS_TRIG_IGNITION);
+        SOUND_START(m_actor, SS_TRIG_IGNITION);
     }
     else
     {
-        SoundScriptManager::getSingleton().trigStop(m_actor->trucknum, SS_TRIG_IGNITION);
+        SOUND_STOP(m_actor, SS_TRIG_IGNITION);
     }
-#endif // USE_OPENAL
 }
 
 void BeamEngine::start()
@@ -1021,10 +1010,8 @@ void BeamEngine::start()
     {
         autoselect = DRIVE;
     }
-#ifdef USE_OPENAL
-    SoundScriptManager::getSingleton().trigStart(m_actor->trucknum, SS_TRIG_IGNITION);
-    SoundScriptManager::getSingleton().trigStart(m_actor->trucknum, SS_TRIG_ENGINE);
-#endif // USE_OPENAL
+    SOUND_START(m_actor, SS_TRIG_IGNITION);
+    SOUND_START(m_actor, SS_TRIG_ENGINE);
 }
 
 void BeamEngine::offstart()
@@ -1084,11 +1071,8 @@ void BeamEngine::stop()
         return;
 
     running = false;
-    // Script Event - engine death
     TRIGGER_EVENT(SE_TRUCK_ENGINE_DIED, m_actor->trucknum);
-#ifdef USE_OPENAL
-    SoundScriptManager::getSingleton().trigStop(m_actor->trucknum, SS_TRIG_ENGINE);
-#endif // USE_OPENAL
+    SOUND_STOP(m_actor, SS_TRIG_ENGINE);
 }
 
 // high level controls
@@ -1115,15 +1099,11 @@ void BeamEngine::shift(int val)
     {
         if (curClutch > 0.25f)
         {
-#ifdef USE_OPENAL
-            SoundScriptManager::getSingleton().trigOnce(m_actor->trucknum, SS_TRIG_GEARSLIDE);
-#endif // USE_OPENAL
+            SOUND_PLAY_ONCE(m_actor, SS_TRIG_GEARSLIDE);
         }
         else
         {
-#ifdef USE_OPENAL
-            SoundScriptManager::getSingleton().trigOnce(m_actor->trucknum, SS_TRIG_SHIFT);
-#endif // USE_OPENAL
+            SOUND_PLAY_ONCE(m_actor, SS_TRIG_SHIFT);
             curGear += val;
         }
     }
@@ -1141,9 +1121,7 @@ void BeamEngine::updateShifts()
     if (autoselect == MANUALMODE)
         return;
 
-#ifdef USE_OPENAL
-    SoundScriptManager::getSingleton().trigOnce(m_actor->trucknum, SS_TRIG_SHIFT);
-#endif // USE_OPENAL
+    SOUND_PLAY_ONCE(m_actor, SS_TRIG_SHIFT);
 
     if (autoselect == REAR)
     {

--- a/source/main/gameplay/LandVehicleSimulation.cpp
+++ b/source/main/gameplay/LandVehicleSimulation.cpp
@@ -318,16 +318,12 @@ void LandVehicleSimulation::UpdateVehicle(Beam* curr_truck, float seconds_since_
             {
                 // starter
                 curr_truck->engine->setstarter(1);
-#ifdef USE_OPENAL
-                SoundScriptManager::getSingleton().trigStart(curr_truck, SS_TRIG_STARTER);
-#endif // OPENAL
+                SOUND_START(curr_truck, SS_TRIG_STARTER);
             }
             else
             {
                 curr_truck->engine->setstarter(0);
-#ifdef USE_OPENAL
-                SoundScriptManager::getSingleton().trigStop(curr_truck, SS_TRIG_STARTER);
-#endif // OPENAL
+                SOUND_STOP(curr_truck, SS_TRIG_STARTER);
             }
 
             if (RoR::App::GetInputEngine()->getEventBoolValueBounce(EV_TRUCK_SWITCH_SHIFT_MODES))
@@ -502,16 +498,14 @@ void LandVehicleSimulation::UpdateVehicle(Beam* curr_truck, float seconds_since_
                 }
             }
         } // end of ->engine
-#ifdef USE_OPENAL
         if (curr_truck->brake > curr_truck->brakeforce / 6.0f)
         {
-            SoundScriptManager::getSingleton().trigStart(curr_truck, SS_TRIG_BRAKE);
+            SOUND_START(curr_truck, SS_TRIG_BRAKE);
         }
         else
         {
-            SoundScriptManager::getSingleton().trigStop(curr_truck, SS_TRIG_BRAKE);
+            SOUND_STOP(curr_truck, SS_TRIG_BRAKE);
         }
-#endif // USE_OPENAL
     } // end of ->replaymode
 
     if (RoR::App::GetInputEngine()->getEventBoolValueBounce(EV_TRUCK_TOGGLE_AXLE_LOCK))
@@ -530,26 +524,24 @@ void LandVehicleSimulation::UpdateVehicle(Beam* curr_truck, float seconds_since_
         }
     }
 
-#ifdef USE_OPENAL
     if (curr_truck->ispolice)
     {
         if (RoR::App::GetInputEngine()->getEventBoolValueBounce(EV_TRUCK_HORN))
         {
-            SoundScriptManager::getSingleton().trigToggle(curr_truck, SS_TRIG_HORN);
+            SOUND_TOGGLE(curr_truck, SS_TRIG_HORN);
         }
     }
     else
     {
         if (RoR::App::GetInputEngine()->getEventBoolValue(EV_TRUCK_HORN) && !curr_truck->replaymode)
         {
-            SoundScriptManager::getSingleton().trigStart(curr_truck, SS_TRIG_HORN);
+            SOUND_START(curr_truck, SS_TRIG_HORN);
         }
         else
         {
-            SoundScriptManager::getSingleton().trigStop(curr_truck, SS_TRIG_HORN);
+            SOUND_STOP(curr_truck, SS_TRIG_HORN);
         }
     }
-#endif // OPENAL
 
     if (RoR::App::GetInputEngine()->getEventBoolValueBounce(EV_TRUCK_PARKING_BRAKE))
     {

--- a/source/main/gameplay/RoRFrameListener.cpp
+++ b/source/main/gameplay/RoRFrameListener.cpp
@@ -1607,7 +1607,7 @@ bool RoRFrameListener::frameStarted(const FrameEvent& evt)
             // calculate orientation of avatar first
             Ogre::Vector3 avatarDir = Ogre::Vector3(Math::Cos(gEnv->player->getRotation()), 0.0f, Math::Sin(gEnv->player->getRotation()));
 
-            SoundScriptManager::getSingleton().GetMumble()->update(gEnv->mainCamera->getPosition(), gEnv->mainCamera->getDirection(), gEnv->mainCamera->getUp(),
+            App::GetMumble()->update(gEnv->mainCamera->getPosition(), gEnv->mainCamera->getDirection(), gEnv->mainCamera->getUp(),
                 gEnv->player->getPosition() + Vector3(0, 1.8f, 0), avatarDir, Ogre::Vector3(0.0f, 1.0f, 0.0f));
         }
 #endif // USE_MUMBLE

--- a/source/main/gameplay/RoRFrameListener.cpp
+++ b/source/main/gameplay/RoRFrameListener.cpp
@@ -1140,9 +1140,8 @@ bool RoRFrameListener::UpdateInputEvents(float dt)
                         {
                             if (RoR::App::GetOverlayWrapper())
                                 RoR::App::GetOverlayWrapper()->showPressureOverlay(true);
-#ifdef USE_OPENAL
-                            SoundScriptManager::getSingleton().trigStart(curr_truck, SS_TRIG_AIR);
-#endif // OPENAL
+
+                            SOUND_START(curr_truck, SS_TRIG_AIR);
                         }
                     }
                     else if (RoR::App::GetInputEngine()->getEventBoolValue(EV_COMMON_PRESSURE_MORE))
@@ -1151,16 +1150,13 @@ bool RoRFrameListener::UpdateInputEvents(float dt)
                         {
                             if (RoR::App::GetOverlayWrapper())
                                 RoR::App::GetOverlayWrapper()->showPressureOverlay(true);
-#ifdef USE_OPENAL
-                            SoundScriptManager::getSingleton().trigStart(curr_truck, SS_TRIG_AIR);
-#endif // OPENAL
+
+                            SOUND_START(curr_truck, SS_TRIG_AIR);
                         }
                     }
                     else if (m_pressure_pressed)
                     {
-#ifdef USE_OPENAL
-                        SoundScriptManager::getSingleton().trigStop(curr_truck, SS_TRIG_AIR);
-#endif // OPENAL
+                        SOUND_STOP(curr_truck, SS_TRIG_AIR);
                         m_pressure_pressed = false;
                         if (RoR::App::GetOverlayWrapper())
                             RoR::App::GetOverlayWrapper()->showPressureOverlay(false);
@@ -2040,10 +2036,8 @@ void RoRFrameListener::ChangedCurrentVehicle(Beam* previous_vehicle, Beam* curre
             RoR::App::GetOverlayWrapper()->showDashboardOverlays(false, previous_vehicle);
         }
 
-#ifdef USE_OPENAL
-        SoundScriptManager::getSingleton().trigStop(previous_vehicle, SS_TRIG_AIR);
-        SoundScriptManager::getSingleton().trigStop(previous_vehicle, SS_TRIG_PUMP);
-#endif // OPENAL
+        SOUND_STOP(previous_vehicle, SS_TRIG_AIR);
+        SOUND_STOP(previous_vehicle, SS_TRIG_PUMP);
     }
 
     if (current_vehicle == nullptr)
@@ -2422,7 +2416,7 @@ bool RoRFrameListener::SetupGameplayLoop()
 
     if (App::audio_menu_music.GetActive())
     {
-        SoundScriptManager::getSingleton().trigKill(-1, SS_TRIG_MAIN_MENU);
+        SOUND_KILL(-1, SS_TRIG_MAIN_MENU);
     }
 
     App::CreateSceneMouse();

--- a/source/main/gui/GUIManager.cpp
+++ b/source/main/gui/GUIManager.cpp
@@ -551,7 +551,9 @@ void GUIManager::DrawMpConnectingStatusBox()
     ImGui::NextColumn();
     // HACK: The trailing space is a workaround for a scissoring issue in OGRE/DearIMGUI integration. ~ only_a_ptr, 10/2017
     ImGui::Text("Joining [%s:%d] ", App::mp_server_host.GetActive(), App::mp_server_port.GetActive());
+#ifdef USE_SOCKETW
     ImGui::TextDisabled(Networking::GetStatusMessage().GetBuffer());
+#endif
     ImGui::End();
 }
 

--- a/source/main/gui/OverlayWrapper.cpp
+++ b/source/main/gui/OverlayWrapper.cpp
@@ -951,13 +951,11 @@ void OverlayWrapper::UpdateAerialHUD(Beam* vehicle)
     if (absangle < 0)
         absangle = -absangle;
 
-#ifdef USE_OPENAL
-    SoundScriptManager::getSingleton().modulate(vehicle, SS_MOD_AOA, absangle);
+    SOUND_MODULATE(vehicle, SS_MOD_AOA, absangle);
     if (absangle > 18.0) // TODO: magicccc
-        SoundScriptManager::getSingleton().trigStart(vehicle, SS_TRIG_AOA);
+        SOUND_START(vehicle, SS_TRIG_AOA);
     else
-        SoundScriptManager::getSingleton().trigStop(vehicle, SS_TRIG_AOA);
-#endif // OPENAL
+        SOUND_STOP(vehicle, SS_TRIG_AOA);
 
     if (angle > 25.0)
         angle = 25.0;

--- a/source/main/main.cpp
+++ b/source/main/main.cpp
@@ -302,9 +302,8 @@ int main(int argc, char *argv[])
                     menu_wallpaper_widget->setVisible(true);
 
 #ifdef USE_MUMBLE
-                    auto* mumble = SoundScriptManager::getSingleton().GetMumble();
-                    if (mumble != nullptr)
-                        mumble->SetNonPositionalAudio();
+                    if (App::GetMumble() != nullptr)
+                        App::GetMumble()->SetNonPositionalAudio();
 #endif // USE_MUMBLE
                 }
 

--- a/source/main/main.cpp
+++ b/source/main/main.cpp
@@ -307,11 +307,13 @@ int main(int argc, char *argv[])
 #endif // USE_MUMBLE
                 }
 
+#ifdef USE_OPENAL
                 if (App::audio_menu_music.GetActive())
                 {
                     SoundScriptManager::getSingleton().createInstance("tracks/main_menu_tune", -1, nullptr);
-                    SoundScriptManager::getSingleton().trigStart(-1, SS_TRIG_MAIN_MENU);
+                    SOUND_START(-1, SS_TRIG_MAIN_MENU);
                 }
+#endif // USE_OPENAL
 
                 App::GetGuiManager()->ReflectGameState();
                 if (App::mp_state.GetPending() == MpState::CONNECTED || BSETTING("SkipMainMenu", false))

--- a/source/main/physics/Beam.cpp
+++ b/source/main/physics/Beam.cpp
@@ -116,10 +116,10 @@ Beam::~Beam()
 #ifdef USE_OPENAL
     for (int i = SS_TRIG_NONE + 1; i < SS_MAX_TRIG; i++)
     {
-        SoundScriptManager::getSingleton().trigStop(this->trucknum, i);
+        SOUND_STOP(this->trucknum, i);
     }
+#endif
     StopAllSounds();
-#endif // USE_OPENAL
 
     // destruct and remove every tiny bit of stuff we created :-|
     if (engine)
@@ -758,19 +758,17 @@ void Beam::calcNetwork()
     int gear = oob1->engine_gear;
     unsigned int flagmask = oob1->flagmask;
 
-#ifdef USE_OPENAL
     if (engine)
     {
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_ENGINE, engspeed);
+        SOUND_MODULATE(trucknum, SS_MOD_ENGINE, engspeed);
     }
     if (free_aeroengine > 0)
     {
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_AEROENGINE1, engspeed);
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_AEROENGINE2, engspeed);
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_AEROENGINE3, engspeed);
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_AEROENGINE4, engspeed);
+        SOUND_MODULATE(trucknum, SS_MOD_AEROENGINE1, engspeed);
+        SOUND_MODULATE(trucknum, SS_MOD_AEROENGINE2, engspeed);
+        SOUND_MODULATE(trucknum, SS_MOD_AEROENGINE3, engspeed);
+        SOUND_MODULATE(trucknum, SS_MOD_AEROENGINE4, engspeed);
     }
-#endif //OPENAL
 
     brake = netbrake;
 
@@ -820,17 +818,15 @@ void Beam::calcNetwork()
     netBrakeLight = ((flagmask & NETMASK_BRAKES) != 0);
     netReverseLight = ((flagmask & NETMASK_REVERSE) != 0);
 
-#ifdef USE_OPENAL
     if ((flagmask & NETMASK_HORN))
-        SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_HORN);
+        SOUND_START(trucknum, SS_TRIG_HORN);
     else
-        SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_HORN);
+        SOUND_STOP(trucknum, SS_TRIG_HORN);
 
     if (netReverseLight)
-        SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_REVERSE_GEAR);
+        SOUND_START(trucknum, SS_TRIG_REVERSE_GEAR);
     else
-        SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_REVERSE_GEAR);
-#endif //OPENAL
+        SOUND_STOP(trucknum, SS_TRIG_REVERSE_GEAR);
 
     updateDashBoards(tratio);
 
@@ -1951,10 +1947,8 @@ void Beam::sendStreamData()
         if (antilockbrake)
             send_oob->flagmask += NETMASK_ALB_ACTIVE;
 
-#ifdef USE_OPENAL
-        if (SoundScriptManager::getSingleton().getTrigState(trucknum, SS_TRIG_HORN))
+        if (SOUND_GET_STATE(trucknum, SS_TRIG_HORN))
             send_oob->flagmask += NETMASK_HORN;
-#endif //OPENAL
     }
 
     // then process the contents
@@ -3267,28 +3261,28 @@ void Beam::updateFlares(float dt, bool isCurrent)
         if (flares[i].type == 'l' && blinkingtype == BLINK_LEFT)
         {
             left_blink_on = isvisible;
-#ifdef USE_OPENAL
+
             if (left_blink_on)
-                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_TURN_SIGNAL_TICK);
-#endif //USE_OPENAL
+                SOUND_PLAY_ONCE(trucknum, SS_TRIG_TURN_SIGNAL_TICK);
+
             dash->setBool(DD_SIGNAL_TURNLEFT, isvisible);
         }
         else if (flares[i].type == 'r' && blinkingtype == BLINK_RIGHT)
         {
             right_blink_on = isvisible;
-#ifdef USE_OPENAL
+
             if (right_blink_on)
-                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_TURN_SIGNAL_TICK);
-#endif //USE_OPENAL
+                SOUND_PLAY_ONCE(trucknum, SS_TRIG_TURN_SIGNAL_TICK);
+
             dash->setBool(DD_SIGNAL_TURNRIGHT, isvisible);
         }
         else if (flares[i].type == 'l' && blinkingtype == BLINK_WARN)
         {
             warn_blink_on = isvisible;
-#ifdef USE_OPENAL
+
             if (warn_blink_on)
-                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_TURN_SIGNAL_WARN_TICK);
-#endif //USE_OPENAL
+                SOUND_PLAY_ONCE(trucknum, SS_TRIG_TURN_SIGNAL_WARN_TICK);
+
             dash->setBool(DD_SIGNAL_TURNRIGHT, isvisible);
             dash->setBool(DD_SIGNAL_TURNLEFT, isvisible);
         }
@@ -3356,16 +3350,14 @@ void Beam::setBlinkType(blinktype blink)
     right_blink_on = false;
     warn_blink_on = false;
 
-#ifdef USE_OPENAL
     if (blink == BLINK_NONE)
     {
-        SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_TURN_SIGNAL);
+        SOUND_STOP(trucknum, SS_TRIG_TURN_SIGNAL);
     }
     else
     {
-        SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_TURN_SIGNAL);
+        SOUND_START(trucknum, SS_TRIG_TURN_SIGNAL);
     }
-#endif //OPENAL
 }
 
 void Beam::autoBlinkReset()
@@ -3467,8 +3459,8 @@ void Beam::updateSoundSources()
         soundsources[i].ssi->setPosition(nodes[soundsources[i].nodenum].AbsPosition, nodes[soundsources[i].nodenum].Velocity);
     }
     //also this, so it is updated always, and for any vehicle
-    SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_AIRSPEED, nodes[0].Velocity.length() * 1.9438);
-    SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_WHEELSPEED, WheelSpeed * 3.6);
+    SOUND_MODULATE(trucknum, SS_MOD_AIRSPEED, nodes[0].Velocity.length() * 1.9438);
+    SOUND_MODULATE(trucknum, SS_MOD_WHEELSPEED, WheelSpeed * 3.6);
 #endif //OPENAL
     BES_GFX_STOP(BES_GFX_updateSoundSources);
 }
@@ -3582,7 +3574,7 @@ void Beam::updateVisual(float dt)
         avichatter_timer -= dt;
         if (avichatter_timer < 0)
         {
-            SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_AVICHAT01 + Math::RangeRandom(0, 12));
+            SOUND_PLAY_ONCE(trucknum, SS_TRIG_AVICHAT01 + Math::RangeRandom(0, 12));
             avichatter_timer = Math::RangeRandom(11, 30);
         }
     }
@@ -4561,12 +4553,10 @@ void Beam::parkingbrakeToggle()
 {
     parkingbrake = !parkingbrake;
 
-#ifdef USE_OPENAL
     if (parkingbrake)
-        SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_PARK);
+        SOUND_START(trucknum, SS_TRIG_PARK);
     else
-        SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_PARK);
-#endif // USE_OPENAL
+        SOUND_STOP(trucknum, SS_TRIG_PARK);
 
     //ScriptEvent - Parking Brake toggle
     TRIGGER_EVENT(SE_TRUCK_PARKINGBREAK_TOGGLE, trucknum);
@@ -4948,20 +4938,24 @@ bool Beam::getReverseLightVisible()
 
 void Beam::StopAllSounds()
 {
+#ifdef USE_OPENAL
     for (int i = 0; i < free_soundsource; i++)
     {
         if (soundsources[i].ssi)
             soundsources[i].ssi->setEnabled(false);
     }
+#endif // USE_OPENAL
 }
 
 void Beam::UnmuteAllSounds()
 {
+#ifdef USE_OPENAL
     for (int i = 0; i < free_soundsource; i++)
     {
         bool enabled = (soundsources[i].type == -2 || soundsources[i].type == currentcamera);
         soundsources[i].ssi->setEnabled(enabled);
     }
+#endif // USE_OPENAL
 }
 
 void Beam::changedCamera()

--- a/source/main/physics/BeamFactory.cpp
+++ b/source/main/physics/BeamFactory.cpp
@@ -747,9 +747,8 @@ void BeamFactory::repairTruck(Collisions* collisions, const Ogre::String& inst, 
     if (rtruck >= 0)
     {
         // take a position reference
-#ifdef USE_OPENAL
-        SoundScriptManager::getSingleton().trigOnce(rtruck, SS_TRIG_REPAIR);
-#endif // USE_OPENAL
+        SOUND_PLAY_ONCE(rtruck, SS_TRIG_REPAIR);
+
         Vector3 ipos = m_trucks[rtruck]->nodes[0].AbsPosition;
         m_trucks[rtruck]->reset();
         m_trucks[rtruck]->resetPosition(ipos.x, ipos.z, false, 0);

--- a/source/main/physics/BeamForcesEuler.cpp
+++ b/source/main/physics/BeamForcesEuler.cpp
@@ -606,28 +606,20 @@ void Beam::calcForcesEulerCompute(int doUpdate, Real dt, int step, int maxsteps)
     {
         if (!antilockbrake)
         {
-#ifdef USE_OPENAL
-            SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_ALB_ACTIVE);
-#endif //USE_OPENAL
+            SOUND_STOP(trucknum, SS_TRIG_ALB_ACTIVE);
         }
         else
         {
-#ifdef USE_OPENAL
-            SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_ALB_ACTIVE);
-#endif //USE_OPENAL
+            SOUND_START(trucknum, SS_TRIG_ALB_ACTIVE);
         }
 
         if (!tractioncontrol)
         {
-#ifdef USE_OPENAL
-            SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_TC_ACTIVE);
-#endif //USE_OPENAL
+            SOUND_STOP(trucknum, SS_TRIG_TC_ACTIVE);
         }
         else
         {
-#ifdef USE_OPENAL
-            SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_TC_ACTIVE);
-#endif //USE_OPENAL
+            SOUND_START(trucknum, SS_TRIG_TC_ACTIVE);
         }
     }
 
@@ -705,12 +697,10 @@ void Beam::calcForcesEulerCompute(int doUpdate, Real dt, int step, int maxsteps)
             stabcommand = 0;
         }
 
-#ifdef USE_OPENAL
         if (stabcommand && fabs(stabratio) < 0.1)
-            SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_AIR);
+            SOUND_START(trucknum, SS_TRIG_AIR);
         else
-            SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_AIR);
-#endif //OPENAL
+            SOUND_STOP(trucknum, SS_TRIG_AIR);
     }
 
     BES_STOP(BES_CORE_Shocks);
@@ -1161,13 +1151,13 @@ void Beam::calcForcesEulerCompute(int doUpdate, Real dt, int step, int maxsteps)
 #ifdef USE_OPENAL
             if (active > 0)
             {
-                SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_PUMP);
+                SOUND_START(trucknum, SS_TRIG_PUMP);
                 float pump_rpm = 660.0f * (1.0f - (work / (float)active) / 100.0f);
-                SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_PUMP, pump_rpm);
+                SOUND_MODULATE(trucknum, SS_MOD_PUMP, pump_rpm);
             }
             else
             {
-                SoundScriptManager::getSingleton().trigStop(trucknum, SS_TRIG_PUMP);
+                SOUND_STOP(trucknum, SS_TRIG_PUMP);
             }
 #endif //USE_OPENAL
         }
@@ -1468,10 +1458,9 @@ void Beam::calcBeams(int doUpdate, Ogre::Real dt, int step, int maxsteps)
                 {
                     // Sound effect.
                     // Sound volume depends on springs stored energy
-#ifdef USE_OPENAL
-                    SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_BREAK, 0.5 * k * difftoBeamL * difftoBeamL);
-                    SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_BREAK);
-#endif //OPENAL
+                    SOUND_MODULATE(trucknum, SS_MOD_BREAK, 0.5 * k * difftoBeamL * difftoBeamL);
+                    SOUND_PLAY_ONCE(trucknum, SS_TRIG_BREAK);
+
                     increased_accuracy = true;
 
                     //Break the beam only when it is not connected to a node
@@ -1639,10 +1628,8 @@ void Beam::calcBeamsInterTruck(int doUpdate, Ogre::Real dt, int step, int maxste
                 {
                     // Sound effect.
                     // Sound volume depends on springs stored energy
-#ifdef USE_OPENAL
-                    SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_BREAK, 0.5 * k * difftoBeamL * difftoBeamL);
-                    SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_BREAK);
-#endif //OPENAL
+                    SOUND_MODULATE(trucknum, SS_MOD_BREAK, 0.5 * k * difftoBeamL * difftoBeamL);
+                    SOUND_PLAY_ONCE(trucknum, SS_TRIG_BREAK);
 
                     //Break the beam only when it is not connected to a node
                     //which is a part of a collision triangle and has 2 "live" beams or less
@@ -1737,10 +1724,10 @@ void Beam::calcNodes(int doUpdate, Ogre::Real dt, int step, int maxsteps)
                             {
                                 if (dustp)
                                     dustp->allocSmoke(nodes[i].AbsPosition, nodes[i].Velocity);
-#ifdef USE_OPENAL
-                                SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_SCREETCH, (ns - thresold) / thresold);
-                                SoundScriptManager::getSingleton().trigOnce(trucknum, SS_TRIG_SCREETCH);
-#endif //USE_OPENAL
+
+                                SOUND_MODULATE(trucknum, SS_MOD_SCREETCH, (ns - thresold) / thresold);
+                                SOUND_PLAY_ONCE(trucknum, SS_TRIG_SCREETCH);
+
                                 //Shouldn't skidmarks be activated from here?
                                 if (useSkidmarks)
                                 {

--- a/source/main/physics/RigSpawner.cpp
+++ b/source/main/physics/RigSpawner.cpp
@@ -1306,6 +1306,7 @@ float RigSpawner::ComputeWingArea(Ogre::Vector3 const & ref, Ogre::Vector3 const
 
 void RigSpawner::ProcessSoundSource2(RigDef::SoundSource2 & def)
 {
+#ifdef USE_OPENAL
     SPAWNER_PROFILE_SCOPED();
 
     int mode = (def.mode == RigDef::SoundSource2::MODE_CINECAM) ? def.cinecam_index : def.mode;
@@ -1320,13 +1321,15 @@ void RigSpawner::ProcessSoundSource2(RigDef::SoundSource2 & def)
             node_index,
             mode
         );
+#endif // USE_OPENAL
 }
 
 void RigSpawner::AddSoundSourceInstance(Beam *vehicle, Ogre::String const & sound_script_name, int node_index, int type)
 {
     SPAWNER_PROFILE_SCOPED();
-
+#ifdef USE_OPENAL
     AddSoundSource(vehicle, SoundScriptManager::getSingleton().createInstance(sound_script_name, vehicle->trucknum, nullptr), node_index);
+#endif // USE_OPENAL
 }
 
 void RigSpawner::AddSoundSource(Beam *vehicle, SoundScriptInstance *sound_script, int node_index, int type)
@@ -1351,6 +1354,7 @@ void RigSpawner::AddSoundSource(Beam *vehicle, SoundScriptInstance *sound_script
 
 void RigSpawner::ProcessSoundSource(RigDef::SoundSource & def)
 {
+#ifdef USE_OPENAL
     SPAWNER_PROFILE_SCOPED();
 
     AddSoundSource(
@@ -1359,6 +1363,7 @@ void RigSpawner::ProcessSoundSource(RigDef::SoundSource & def)
             GetNodeIndexOrThrow(def.node),
             -2
         );
+#endif // USE_OPENAL
 }
 
 void RigSpawner::ProcessCameraRail(RigDef::CameraRail & def)
@@ -6836,8 +6841,8 @@ void RigSpawner::SetupDefaultSoundSources(Beam *vehicle)
         else
             AddSoundSourceInstance(vehicle, "tracks/default_marine_small", smokeId);
         //no start/stop engine for boats, so set sound always on!
-        SoundScriptManager::getSingleton().trigStart(trucknum, SS_TRIG_ENGINE);
-        SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_ENGINE, 0.5);
+        SOUND_START(trucknum, SS_TRIG_ENGINE);
+        SOUND_MODULATE(trucknum, SS_MOD_ENGINE, 0.5);
     }
     //airplane warnings
     if (vehicle->driveable==AIRPLANE)

--- a/source/main/physics/air/TurboJet.cpp
+++ b/source/main/physics/air/TurboJet.cpp
@@ -116,10 +116,10 @@ Turbojet::~Turbojet()
 {
     //A fast work around 
     //
-    SoundScriptManager::getSingleton().modulate(trucknum, thr_id, 0);
-    SoundScriptManager::getSingleton().modulate(trucknum, mod_id, 0);
-    SoundScriptManager::getSingleton().trigStop(trucknum, ab_id);
-    SoundScriptManager::getSingleton().trigStop(trucknum, src_id);
+    SOUND_MODULATE(trucknum, thr_id, 0);
+    SOUND_MODULATE(trucknum, mod_id, 0);
+    SOUND_STOP(trucknum, ab_id);
+    SOUND_STOP(trucknum, src_id);
 
     if (flameMesh != nullptr)
     {
@@ -214,10 +214,7 @@ void Turbojet::updateForces(float dt, int doUpdate)
 {
     if (doUpdate)
     {
-#ifdef USE_OPENAL
-        //sound update
-        SoundScriptManager::getSingleton().modulate(trucknum, mod_id, rpm);
-#endif //OPENAL
+        SOUND_MODULATE(trucknum, mod_id, rpm);
     }
     timer += dt;
     axis = nodes[nodefront].RelPosition - nodes[nodeback].RelPosition;
@@ -258,12 +255,11 @@ void Turbojet::updateForces(float dt, int doUpdate)
     }
     else
         afterburner = false;
-#ifdef USE_OPENAL
+
     if (afterburner)
-        SoundScriptManager::getSingleton().trigStart(trucknum, ab_id);
+        SOUND_START(trucknum, ab_id);
     else
-        SoundScriptManager::getSingleton().trigStop(trucknum, ab_id);
-#endif //OPENAL
+        SOUND_START(trucknum, ab_id);
 
     nodes[nodeback].Forces += (enginethrust * 1000.0) * axis;
     exhaust_velocity = enginethrust * 5.6 / area;
@@ -273,13 +269,12 @@ void Turbojet::setThrottle(float val)
 {
     if (val > 1.0)
         val = 1.0;
+
     if (val < 0.0)
         val = 0.0;
+
     throtle = val;
-#ifdef USE_OPENAL
-    //sound update
-    SoundScriptManager::getSingleton().modulate(trucknum, thr_id, val);
-#endif //OPENAL
+    SOUND_MODULATE(trucknum, thr_id, val);
 }
 
 float Turbojet::getThrottle()
@@ -319,15 +314,11 @@ void Turbojet::flipStart()
     {
         warmup = true;
         warmupstart = timer;
-#ifdef USE_OPENAL
-        SoundScriptManager::getSingleton().trigStart(trucknum, src_id);
-#endif //OPENAL
+        SOUND_START(trucknum, src_id);
     }
     else
     {
-#ifdef USE_OPENAL
-        SoundScriptManager::getSingleton().trigStop(trucknum, src_id);
-#endif //OPENAL
+        SOUND_STOP(trucknum, src_id);
     }
 
     lastflip = timer;

--- a/source/main/physics/air/TurboProp.cpp
+++ b/source/main/physics/air/TurboProp.cpp
@@ -157,9 +157,9 @@ Turboprop::~Turboprop()
 {
     //A fast work around 
     //
-    SoundScriptManager::getSingleton().modulate(trucknum, thr_id, 0);
-    SoundScriptManager::getSingleton().modulate(trucknum, mod_id, 0);
-    SoundScriptManager::getSingleton().trigStop(trucknum, src_id);
+    SOUND_MODULATE(trucknum, thr_id, 0);
+    SOUND_MODULATE(trucknum, mod_id, 0);
+    SOUND_STOP(trucknum, src_id);
 
     if (airfoil != nullptr)
         delete airfoil;
@@ -255,10 +255,7 @@ void Turboprop::updateForces(float dt, int doUpdate)
         //float airtemperature=sea_level_temperature-altitude*0.0065; //in Kelvin
         float airpressure = sea_level_pressure * pow(1.0 - 0.0065 * altitude / 288.15, 5.24947); //in Pa
         airdensity = airpressure * 0.0000120896;//1.225 at sea level
-#ifdef USE_OPENAL
-        //sound update
-        SoundScriptManager::getSingleton().modulate(trucknum, mod_id, rpm);
-#endif //OPENAL
+        SOUND_MODULATE(trucknum, mod_id, rpm);
     }
 
     timer += dt;
@@ -471,10 +468,7 @@ void Turboprop::setThrottle(float val)
     if (val < 0.0)
         val = 0.0;
     throtle = val;
-#ifdef USE_OPENAL
-    //sound update
-    SoundScriptManager::getSingleton().modulate(trucknum, thr_id, val);
-#endif //OPENAL
+    SOUND_MODULATE(trucknum, thr_id, val);
 }
 
 float Turboprop::getThrottle()
@@ -514,15 +508,11 @@ void Turboprop::flipStart()
     {
         warmup = true;
         warmupstart = timer;
-#ifdef USE_OPENAL
-        SoundScriptManager::getSingleton().trigStart(trucknum, src_id);
-#endif //OPENAL
+        SOUND_START(trucknum, src_id);
     }
     else
     {
-#ifdef USE_OPENAL
-        SoundScriptManager::getSingleton().trigStop(trucknum, src_id);
-#endif //OPENAL
+        SOUND_STOP(trucknum, src_id);
     }
     lastflip = timer;
 }

--- a/source/main/physics/water/ScrewProp.cpp
+++ b/source/main/physics/water/ScrewProp.cpp
@@ -78,9 +78,7 @@ void Screwprop::setThrottle(float val)
     reverse = (val < 0);
     //pseudo-rpm
     float prpm = (0.5 + fabs(val) / 2.0) * 100.0;
-#ifdef USE_OPENAL
-    SoundScriptManager::getSingleton().modulate(trucknum, SS_MOD_ENGINE, prpm);
-#endif // USE_OPENAL
+    SOUND_MODULATE(trucknum, SS_MOD_ENGINE, prpm);
 }
 
 void Screwprop::setRudder(float val)

